### PR TITLE
IF: Update for latest IF

### PR DESCRIPTION
**Pull request by heifner**
## Change Description

- Changed tests to use `controller::active_producers()`  instead of accessing `block_header_state` directly.
- Changed to new CDT `finalizer_policy` struct

## Deployment Changes
- [ ] Deployment Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Comment by greg7mdp** [comment]
Is the target branch correct? `merge-main-01-08-2023`

File: tests/eosio.system_tester.hpp [hunk 1]
```diff
@@ -1024,9 +1024,9 @@ class eosio_system_tester : public TESTER {
       }
       produce_blocks( 250 );
 
-      auto producer_keys = control->head_block_state()->active_schedule.producers;
-      BOOST_REQUIRE_EQUAL( 21, producer_keys.size() );
-      BOOST_REQUIRE_EQUAL( name("defproducera"), producer_keys[0].producer_name );
+      auto producer_keys = control->active_producers();
```
Comment (greg7mdp) [inline tests/eosio.system_tester.hpp:1027]: strange that the variable is named `producer_keys` when it is a `producer_authority_schedule`

**Review by greg7mdp — APPROVED** [review]

File: tests/eosio.system_tester.hpp (hunk 1 above)
Comment (heifner) [reply tests/eosio.system_tester.hpp]: Renamed

**Review by greg7mdp — APPROVED** [review]

**Comment by linh2931** [comment]
> Is the target branch correct? `merge-main-01-08-2023`

Or `instant_finality` branch?

**Comment by heifner** [comment]
> > Is the target branch correct? `merge-main-01-08-2023`
> 
> Or `instant_finality` branch?

It is [merge-main-01-08-2023](https://github.com/AntelopeIO/reference-contracts/tree/merge-main-01-08-2023) because that branch is broken without this fix.

**Review by linh2931 — APPROVED** [review]

**Pull request merged**